src/Orleans/Streams/Core/StreamSubscriptionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams.Core
+{
+    internal class StreamSubscriptionManager: IStreamSubscriptionManager
+    {
+        private IStreamPubSub streamPubSub;
+        private StreamPubSubType type;
+        private string streamProviderName;
+        //potential memory optimization on this
+        private ConcurrentDictionary<Guid, StreamSubscription> allSubscriptions;
```

Comment:
then we will need to **add** one method to IStreamPubSub interface to ask it to 
Task<List<GuidId>> GetAllSubscriptions(StreamId streamId);

I like this alternative more, you want me to go with this?